source/posix/socket.c

```diff
@@ -526,7 +526,8 @@ int aws_socket_connect(
         return aws_raise_error(AWS_IO_EVENT_LOOP_ALREADY_ASSIGNED);
     }
 
-    if (socket->state != INIT) {
+    /* UDP sockets jump straight to CONNECT_READ once bind is called */
+    if (socket->state != INIT && socket->options.type != AWS_SOCKET_DGRAM) {
```

Comment:
You should still check for valid state when udp is used.

Reply:
Good call, fixed.